trunner/runners/ARMV7M7Runner.py

```diff
@@ -0,0 +1,169 @@
+#
+# Phoenix-RTOS test runner
+#
+# ARMV7M7 runner
+#
+# Copyright 2021 Phoenix SYstems
+# Authors: Jakub Sarzyński, Damian Loewnau
+#
+
+import time
+import sys
+import select
+
+from pexpect.exceptions import TIMEOUT, EOF
+from trunner.tools.color import Color
+from .common import Psu, Phoenixd, PhoenixdError, PloError, PloTalker, DeviceRunner
+from .common import GPIO, logging, phd_error_msg, rootfs
+
+
+def hostpc_reboot(serial_downloader=False):
+    """Reboots a tested device by a human."""
+    if serial_downloader:
+        print("\n\tPlease change the board's boot mode to serial download, reset the board and press enter")
+    else:
+        print("\n\tPlease change the board's boot mode to internal/external flash if the current mode is different.\n\
+        Next, reset the board and press enter")
+    print("\tNote that, You should press the enter key just after reset button")
+
+    if sys.stdin in select.select([sys.stdin], [], [], 30)[0]:
+        sys.stdin.readline()
+    else:
+        raise RebootError('It took to long to wait for a key pressing')
+
+
+class RebootError(Exception):
+    def __init__(self, message):
+        msg = Color.colorify("REBOOT ERROR:\n", Color.BOLD)
+        msg += str(message) + '\n'
+        super().__init__(msg)
+
+
+class ARMV7M7Runner(DeviceRunner):
+    """This class provides interface to run tests on targets with armv7m7 architecture"""
+
+    status_color = {'BUSY': 'blue', 'SUCCESS': 'green', 'FAIL': 'red'}
+
+    # redefined by target runners
+    SDP = None
+    IMAGE = None
+
+    def __init__(self, port, is_rpi_host):
+        super().__init__(port)
+        self.is_rpi_host = is_rpi_host
+        # default values, redefined by specified target runners
+        self.phoenixd_port = None
+        self.is_cut_power_used = False
+        self.flash_memory = 0
+
+        if self.is_rpi_host:
+            self.reset_gpio = GPIO(17)
+            self.reset_gpio.high()
+            self.power_gpio = GPIO(2)
+            self.power_gpio.high()
+            self.boot_gpio = GPIO(4)
+            # without this init gpio values an rgb led is turned off for a while
```

Comment:
There was a problem, because, the led gpio init was before setting Runner state to `BUSY` (in runner.flash()) (the first screen). So I moved the setting state to `BUSY` to Runner class constructor. But I had to call super().__init__ after gpio init in ARMV7M7 class (the second screen).
![Screenshot from 2021-12-01 12-11-00](https://user-images.githubusercontent.com/77304431/144225228-10a5246d-5a2f-4250-9c43-bc92bdb018d9.png)

![Screenshot from 2021-12-01 12-16-56](https://user-images.githubusercontent.com/77304431/144225251-f541d894-6d41-4879-97b8-0090a8c9498e.png)